pkg/translator/translator.go

```diff
@@ -225,85 +227,90 @@ func (t *Translator) validateConnectionPorts(h *hpsf.HPSF, templateComps map[str
-		}
-	}
-	// now we need to check that there is only one connection from the StartSampling component to a sampler
-	if startSamplingCount == 1 {
-		samplerConnections := 0
-		for _, conn := range h.Connections {
-			if conn.Source.GetSafeName() == startSamplingComp {
-				// check if the destination is a sampler
-				dstComp, ok := templateComps[conn.Destination.GetSafeName()]
-				if !ok {
-					continue
-				}
-				if dstComp.Style == "sampler" {
-					samplerConnections++
-				}
-			}
-		}
-		if samplerConnections != 1 {
-			err := hpsf.NewError("StartSampling component must have exactly one connection to a sampler").
-				WithComponent(startSamplingComp)
-			result.Add(err)
-		}
-	}
+	// startSamplingCount := 0
+	// var startSamplingComp string
+	// for _, c := range h.Components {
+	// 	tmpl, ok := templateComps[c.GetSafeName()]
+	// 	if !ok {
+	// 		continue
+	// 	}
+
+	// 	if tmpl.Style == "startsampling" {
+	// 		startSamplingCount++
+	// 		startSamplingComp = c.GetSafeName()
+	// 		if startSamplingCount > 1 {
+	// 			err := hpsf.NewError("only one StartSampling component is allowed").
+	// 				WithComponent(c.Name)
+	// 			result.Add(err)
+	// 		}
+	// 	}
+	// }
+	// if startSamplingCount == 0 {
+	// 	// if there is no StartSampling component, we cannot have any samplers in the configuration
+	// 	for _, c := range h.Components {
+	// 		tmpl, ok := templateComps[c.GetSafeName()]
+	// 		if !ok {
+	// 			continue
+	// 		}
+
+	// 		if tmpl.Style == "sampler" {
+	// 			err := hpsf.NewError("if there is no StartSampling component, no samplers are allowed").
+	// 				WithComponent(c.Name)
+	// 			result.Add(err)
+	// 		}
+	// 	}
+	// } else {
+	// 	// if there is a StartSampling component, we must have at least one sampler in the configuration
+	// 	hasSampler := false
+	// 	for _, c := range h.Components {
+	// 		tmpl, ok := templateComps[c.GetSafeName()]
+	// 		if !ok {
+	// 			continue
+	// 		}
+
+	// 		if tmpl.Style == "sampler" {
+	// 			hasSampler = true
+	// 			break
+	// 		}
+	// 	}
+	// 	if !hasSampler {
+	// 		err := hpsf.NewError("if there is a StartSampling component, at least one sampler is required").
+	// 			WithComponent("StartSampling")
+	// 		result.Add(err)
+	// 	}
+	// }
+	// // now we need to check that there is only one connection from the StartSampling component to a sampler
+	// if startSamplingCount == 1 {
+	// 	samplerConnections := 0
+	// 	for _, conn := range h.Connections {
+	// 		if conn.Source.GetSafeName() == startSamplingComp {
+	// 			// check if the destination is a sampler
+	// 			dstComp, ok := templateComps[conn.Destination.GetSafeName()]
+	// 			if !ok {
+	// 				continue
+	// 			}
+	// 			if dstComp.Style == "sampler" {
+	// 				samplerConnections++
+	// 			}
+	// 		}
+	// 	}
+	// 	if samplerConnections != 1 {
+	// 		err := hpsf.NewError("StartSampling component must have exactly one connection to a sampler").
+	// 			WithComponent(startSamplingComp)
+	// 		result.Add(err)
+	// 	}
+	// }
```

Comment:
Do we still need this code?

Reply:
I left this in deliberately because this PR doesn't include fixing all the validation, but this code is still useful as reference.